plugins/modules/idrac_diagnostics.py

```diff
@@ -0,0 +1,818 @@
+
+MANAGERS_URI = "/redfish/v1/Managers"
+
+OEM = "Oem"
+MANUFACTURER = "Dell"
+JOBS = "Jobs"
+JOBS_EXPAND = "?$expand=*($levels=1)"
+LC_SERVICE = "DellLCService"
+ACTIONS = "Actions"
+EXPORT = "#DellLCService.ExportePSADiagnosticsResult"
+RUN = "#DellLCService.RunePSADiagnostics"
+ODATA_REGEX = "(.*?)@odata"
+ODATA = "@odata.id"
+TIME_FORMAT_FILE = "%Y%m%d_%H%M%S"
+TIME_FORMAT_WITHOUT_OFFSET = "%Y%m%d%H%M%S"
+TIME_FORMAT_WITH_OFFSET = "%Y-%m-%dT%H:%M:%S%z"
+SUCCESS_EXPORT_MSG = "Successfully exported the diagnostics."
+SUCCESS_RUN_MSG = "Successfully run the diagnostics operation."
+SUCCESS_RUN_AND_EXPORT_MSG = "Successfully run and exported the diagnostics."
+RUNNING_RUN_MSG = "Successfully triggered the job to run diagnostics."
+ALREADY_RUN_MSG = "The diagnostics job is already present."
+INVALID_FILE_MSG = "File extension is invalid. Supported extension is: .txt."
+INVALID_DIRECTORY_MSG = "Provided directory path '{path}' is not valid."
+NO_OPERATION_SKIP_MSG = "Task is skipped as none of run, export or run and export is specified."
+INSUFFICIENT_DIRECTORY_PERMISSION_MSG = "Provided directory path '{path}' is not writable. " \
+                                        "Please check if the directory has appropriate permissions"
+UNSUPPORTED_FIRMWARE_MSG = "iDRAC firmware version is not supported."
+TIMEOUT_NEGATIVE_OR_ZERO_MSG = "The parameter `job_wait_timeout` value cannot be negative or zero."
+WAIT_TIMEOUT_MSG = "The job is not complete after {0} seconds."
+START_TIME = "The specified scheduled start time occurs in the past, " \
+             "provide a future time to schedule the start time."
+INVALID_TIME = "The specified date and time `{0}` to schedule the diagnostics is not valid. Enter a valid date and time."
+END_START_TIME = "The end time `{0}` to schedule the diagnostics must be greater than the start time `{1}`."
+NO_CHANGES_FOUND_MSG = "No changes found to be applied."
+CHANGES_FOUND_MSG = "Changes found to be applied."
+
+PROXY_SUPPORT = {"off": "Off", "default_proxy": "DefaultProxy", "parameters_proxy": "ParametersProxy"}
+
+
+class RunDiagnostics:
+    STATUS_SUCCESS = [200, 202]
+
+    def __init__(self, idrac, module):
+        self.idrac = idrac
+        self.module = module
+        self.run_url = None
+
+    def execute(self):
+        self.__get_run_diagnostics_url()
+        self.check_diagnostics_jobs()
+        run_diagnostics_status = self.__run_diagnostics()
+        job_status = self.__perform_job_wait(run_diagnostics_status)
+        status = run_diagnostics_status.status_code
+        if status in self.STATUS_SUCCESS and job_status.get('JobState') == "Completed":
+            msg = SUCCESS_RUN_MSG
+            job_details = job_status
+            return msg, job_details
+        if status in self.STATUS_SUCCESS and job_status.get('JobState') in ["Scheduled", "Running", "New"]:
+            msg = RUNNING_RUN_MSG
+            job_details = job_status
+            return msg, job_details
```

Comment:
can be returned outside if else ladder. Only msg can be assigned in if statements and return can be done only once

Reply:
done